query.go

```diff
@@ -782,7 +784,33 @@ func (s *ChainService) GetCFilter(blockHash chainhash.Hash,
 					numFilters)
 			}
 
-			if s.persistToDisk {
+			persistChan <- resp
+		}
+	}()
+
+	if s.persistToDisk {
+		// Persisting to disk is the bottleneck for fetching filters.
+		// So we run the persisting logic in a separate goroutine so
+		// that we can unlock the mtxCFilter mutex as soon as we are
+		// done with caching the filters in order to allow more
+		// GetCFilter calls from the caller sooner.
+		go func() {
+			var (
+				resp *filterResponse
+				ok   bool
+			)
+
+			for {
+				select {
+				case resp, ok = <-persistChan:
+					if !ok {
+						return
+					}
+
+				case <-s.quit:
+					return
+				}
+
```

Comment:
One worry with this is that we spin up like 700 goroutines (on mainnet). So another option to rate limit this a bit is to have a single outer goroutine that listens on a channel with a large buffer. Then we rate limit ourselves with that buffered channel